frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -1421,11 +1421,17 @@ export const surveyLogic = kea<surveyLogicType>([
             (survey: Survey): IntervalType => {
                 const start = getSurveyStartDateForQuery(survey)
                 const end = getSurveyEndDateForQuery(survey)
-                const diffInWeeks = dayjs(end).diff(dayjs(start), 'weeks')
+                const diffInDays = dayjs(end).diff(dayjs(start), 'days')
 
-                if (diffInWeeks <= 4) {
+                if (diffInDays < 2) {
+                    return 'hour'
+                }
+                // less than a month
+                if (diffInDays < 30) {
                     return 'day'
-                } else if (diffInWeeks <= 12) {
+                }
+                // more than a month, less than 3 months
+                if (diffInDays < 90) {
```

Comment:
I'd probably keep the rest of the logic the same, 30 days is different than 4 weeks, at least for a few months like February that has usually 28 days, or months with 31 days
so just do `const diffInWeeks = dayjs(end).diff(dayjs(start), 'weeks')` and keep the rest the same? wdyt?

Reply:
i used 30 days because that's the definition of a ["commercial month"](https://www.investopedia.com/terms/c/commercial-year.asp). if it's less than 30 days, it'll just show the data from the two months, which I don't think matters too much

Reply:
well, it changes the current behavior, and it has more room for a bad UX than the current implementation IMO